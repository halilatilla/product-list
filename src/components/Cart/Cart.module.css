.cart {
  --cart-width: 118px;

  width: var(--cart-width);
  height: 48px;
  border: 1px solid var(--c-border);
  border-radius: 4px;
  position: relative;
  cursor: pointer;
}

.cartTitle {
  font-size: 17px;
  text-transform: capitalize;
  color: var(--c-text-light);
}

.cartDetail {
  /* display: none; */
  position: absolute;
  top: 96%;
  right: -1px;
  border: 1px solid var(--c-border);
  background-color: var(--c-white);
  width: 360px;
  min-height: 100px;
  border-radius: 4px;
  padding: 20px;
}

.cartDetail::after {
  content: '';
  background-color: var(--c-white);
  width: calc(var(--cart-width) - 2px);
  height: 1px;
  position: absolute;
  bottom: 100%;
  right: 0px;
}

.cart:hover {
  border-bottom-color: transparent;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.cart:hover .cartDetail {
  display: block;
  border-top-right-radius: 0;
}

.cartItemList {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.cartItemsCount {
  position: absolute;
  width: 18px;
  height: 18px;
  right: -10px;
  top: -10px;
  border-radius: 50%;
  background-color: var(--c-primary);
  color: var(--c-white);
  font-size: 12px;
  font-weight: 500;
  line-height: 0;
}
