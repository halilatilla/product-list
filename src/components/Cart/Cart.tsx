import { FC, useEffect } from 'react'
import classnames from 'classnames'

import { useAppSelector, useAppDispatch, updateCartItems } from '@src/store'
import { useLocalStorage } from '@src/hooks'
import CartItem from './CartItem/CartItem'
import ICart from './Cart.types'
import styles from './Cart.module.scss'

const Cart: FC<ICart> = ({ className, ...rest }) => {
  const cartItems = useAppSelector((state) => state.cart)
  const dispatch = useAppDispatch()
  const [localCartItems, setLocalCartItems] = useLocalStorage('cart', cartItems)

  useEffect(() => {
    if (localCartItems) {
      dispatch(updateCartItems(localCartItems))
    }
  }, [])

  useEffect(() => {
    setLocalCartItems(cartItems)
  }, [cartItems])

  return (
    <div className={classnames(styles.cart, 'flex-center', className)} {...rest} data-testid="cart">
      {localCartItems?.length > 0 && (
        <div className={classnames(styles.cartItemsCount, 'flex-center')}>{localCartItems?.length}</div>
      )}

      <p className={styles.cartTitle}>sepetim</p>
      <div className={styles.cartDetail}>
        {localCartItems?.length > 0 ? (
          <div className={styles.cartItemList}>
            {localCartItems?.map((item) => (
              <CartItem key={item.productId} item={item} />
            ))}
          </div>
        ) : (
          <div>Sepetinizde ürün bulunmamaktadır.</div>
        )}
      </div>
    </div>
  )
}

export default Cart
