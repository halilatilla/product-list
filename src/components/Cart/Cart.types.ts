export default interface ICart {
  className?: string
}
