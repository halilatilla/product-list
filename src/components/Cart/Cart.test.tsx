import { render } from '@src/test/test-utils'

import Cart from './Cart'

describe('Cart', () => {
  it('renders correctly', () => {
    const { container } = render(<Cart />)
    expect(container).toMatchSnapshot()
  })
})
