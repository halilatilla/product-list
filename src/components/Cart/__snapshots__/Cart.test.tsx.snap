// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly 1`] = `
<div>
  <div
    class="cart flex-center"
  >
    <p
      class="cartTitle"
    >
      sepetim
    </p>
    <div
      class="cartDetail"
    >
      <div>
        Sepetinizde ürün bulunmamaktadır.
      </div>
    </div>
  </div>
</div>
`;
